library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -916,9 +927,22 @@ protected void onInputFormatChanged(Format newFormat) throws ExoPlaybackExceptio
         releaseCodec();
         maybeInitCodec();
       }
+    } else {
+      if (Util.SDK_INT >= 23) {
+        updateCodecOperatingRate(format);
+      }
     }
   }
 
+  /**
+   * Updates the {@link MediaCodec} operating rate.
+   * <p>
+   * The default implementation is a no-op.
+   */
+  protected void updateCodecOperatingRate(Format format) {
```

Comment:
Can we explicitly pass both the codec instance and the new operating rate to this method, so the subclass doesn't need to retrieve it with `getCodec`? It's clearer what's going on, and is more consistent with the `configureCodec` method.